### PR TITLE
Release 0.6

### PR DESCRIPTION
**Pull request by SimonHeybrock**
Preparations for release, not ready yet.

### CODE REVIEW

**Comment by OwenArnold** [comment]
We need to describe the deployment procedure like we have added for `mantid-framework` https://github.com/scipp/scipp/blob/main/docs/developer/deployment.rst Though maybe that makes sense after we've run through it successfully?

**Comment by SimonHeybrock** [comment]
> We need to describe the deployment procedure like we have added for `mantid-framework` https://github.com/scipp/scipp/blob/main/docs/developer/deployment.rst Though maybe that makes sense after we've run through it successfully?

Is this related to the release?

**Comment by OwenArnold** [comment]
> > We need to describe the deployment procedure like we have added for `mantid-framework` https://github.com/scipp/scipp/blob/main/docs/developer/deployment.rst Though maybe that makes sense after we've run through it successfully?
> 
> Is this related to the release?

Not really. That can be done separately.

File: docs/index.rst [hunk 1]
```diff
@@ -42,8 +42,8 @@ If your use case requires one or several of the items on the following list, usi
 - Support for **masks stored with data**.
 - Written in C++ for better performance (for certain applications), in combination with Python bindings.
 
-Generic functionality of scipp is provided in the **scipp** Python module.
-In addition, more specific functionality is made available in other modules.
+Generic functionality of scipp is provided in the **scipp** Python package.
+In addition, more specific functionality is made available in other packes.
```
Comment (nvaytet) [inline docs/index.rst:46]:
```suggestion
In addition, more specific functionality is made available in other packages.
```

File: docs/getting-started/installation.rst [hunk 2]
```diff
@@ -9,28 +9,62 @@ Conda
...
+1. Download :download:`scipp.yml <../environments/scipp.yml>` for the stable release version of scipp.
+   Use :download:`scipp-latest.yml <../environments/scipp-latest.yml>` for the latest development version.
+2. In a terminal run:
+
+   .. code-block:: sh
+
+      conda activate
+      conda env create -f scipp.yml
+      conda activate scipp
+      jupyter lab
```
Comment (nvaytet) [inline docs/getting-started/installation.rst:27]: :+1: for adding the launch command for jupyter.

**Review by nvaytet — CHANGES_REQUESTED** [review]

**Comment by SimonHeybrock** [comment]
As discussed, I added a "What's new" notebook *including* outputs. Since this will depend on a particular released version we cannot auto-generate those in CI easily. The notebooks combined scipp and scippneutron updates. I think this is probably best for now (this release), in the future we may want to switch to having separate ones.

File: docs/about/whats-new/whats-new-0.6.0.ipynb [hunk 3]
```diff
@@ -0,0 +1,6969 @@
...
+       "  <scipp.Variable>          datetime64  [s]              (time)  [2021-01-01T12:02:03, 2021-01-01T12:02:04, ..., 2021-01-01T12:06:01, 2021-01-01T12:06:02]"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "var = sc.array(dims=['time'], values=np.arange(np.datetime64('2021-01-01T12:00:00'), np.datetime64('2021-01-01T12:04:00')))\n",
```
Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:586]: i would add a new line between the two `np.datetime` as the line is a little long when rendered in the notebook.

File: docs/about/whats-new/whats-new-0.6.0.ipynb [hunk 4]
```diff
@@ -0,0 +1,6969 @@
...
+    "sc.table(data.bins.constituents['data']['event', :5])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### HTML view readability improvements\n",
+    "\n",
+    "Readability for nested scipp objects such as `DataArray` in HTML view improved, see in particular the many scalar attributes:"
```
Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:672]:
```suggestion
    "The readability of nested scipp objects such as `DataArray` in the HTML view has been improved, see in particular the many scalar attributes:"
```

File: docs/about/whats-new/whats-new-0.6.0.ipynb [hunk 5]
```diff
@@ -0,0 +1,6969 @@
...
+       "\n",
+       ".sc-attrs dt {\n",
+       "  font-weight: normal;\n",
+       "  grid-column: 1;\n",
+       "}\n",
+       "\n",
+       ".sc-attrs dt:hover span {\n",
+       "  display: inline-block;\n",
+       "  background: #fff;\n",
+
```
Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:3890]:
```suggestion
    "The `bins` property now provides properties `data`, `coords`, `masks`, and `attrs` that behave very similarly to that of a data array.\n",
```

Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:3891]:
```suggestion
    "We can use this, e.g., to compute the time an event passed the sample, and store it as a new coord:"
```

Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:3943]:
```suggestion
    "  For `scipp.neutron.diffraction.convert_with_calibration` see [Beamline geometry parameters for unit conversions](Beamline-geometry-parameters-for-unit-conversions) below for the first iteration of a better approach which could be extented to support something similar in a more flexible manner.\n",
```

Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:3944]: Combine this line with the one below? i.e.

from_mantid (and by extension load) have been improved to support loading of McStasNexus files and WorkspaceGroup workspaces.

Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:3947]: "this case" or "these cases"?

**Review by nvaytet — CHANGES_REQUESTED** [review]

**Comment by nvaytet** [comment]
Can i suggest next time that you only include the output in the notebook after we have reviewed?
It was painful to go through that giant file.

File: docs/about/whats-new/whats-new-0.6.0.ipynb (hunk 5 above)
Comment (SimonHeybrock) [reply docs/about/whats-new/whats-new-0.6.0.ipynb:3890]: Reformulated. 👍

Comment (SimonHeybrock) [reply docs/about/whats-new/whats-new-0.6.0.ipynb:3891]: Can't see a change here? The sentence is the same as before?

Comment (SimonHeybrock) [reply docs/about/whats-new/whats-new-0.6.0.ipynb:3943]: Canno see a change

Comment (SimonHeybrock) [reply docs/about/whats-new/whats-new-0.6.0.ipynb:3944]: 👍

File: docs/about/whats-new/whats-new-0.6.0.ipynb [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Various\n",
+    "\n",
+    "- As a small breaking change, the naming of metadata was changed to use `_` instead of `-`, e.g., `pulse_time`, `sample_position`, and `incident_energy`.\n",
+    "- The contents of (previously) `scipp.neutron.diffraction` are not available any more, since they were considered as too specialized (facility specific).\n",
+    "  For `scipp.neutron.diffraction.convert_with_calibration` see [Beamline geometry parameters for unit conversions](Beamline-geometry-parameters-for-unit-conversions) below for the first iteration of a better approach which could be extented to support something similar in a more flexibly manner.\n",
```
Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:266]:
```suggestion
    "  For `scipp.neutron.diffraction.convert_with_calibration` see [Beamline geometry parameters for unit conversions](Beamline-geometry-parameters-for-unit-conversions) below for the first iteration of a better approach which could be extented to support something similar in a more flexible manner.\n",
```

File: docs/about/whats-new/whats-new-0.6.0.ipynb [hunk 7]
```diff
@@ -0,0 +1,358 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This could be used, e.g., to access the `tof` coordinate of the events.\n",
+    "However, this does not give information on which bin (pixel in this case) an event belongs to, so this is not sufficient if, e.g., a computation involving beamline geometry should be performed.\n",
+    "\n",
+    "To allow for this, we added support for access to the coords (and data, masks, and attrs) of the bins while retaining the binned structure.\n",
+    "The `bins` property now provides properties `data`, `coords`, `masks`, and `attrs` that behave like the properties of a data array.\n",
+    "We can use this, e.g., to compute the time an event passed the sample, and store it is a new coord:"
```
Comment (nvaytet) [inline docs/about/whats-new/whats-new-0.6.0.ipynb:229]:
```suggestion
    "We can use this, e.g., to compute the time an event passed the sample, and store it as a new coord:"
```

**Review by nvaytet — CHANGES_REQUESTED** [review]
small typos

**Comment by SimonHeybrock** [comment]
@nvaytet As per your suggestion, I removed the outputs of the "What's new" notebooks from the repo. The plan is to upload them manually to https://github.com/scipp/scipp.github.io. I have edited `documentation_deploy.yml` to not remove files from this directory. Can't really test this before merging, so will probably need some fixes.

**Comment by SimonHeybrock** [comment]
Not so sure any more if I like what I said above. The mechanism would be:

- Create environment with required scipp version.
- Run the "What's New" notebook.
- Save with outputs.
- Run `make docs` to generate `html`.
- Add, commit, and push to https://github.com/scipp/scipp.github.io
- **Hope that nothing accidentally removes it**, otherwise, go back to square 1.

**Comment by SimonHeybrock** [comment]
Merging, so I can try this out.

**Comment by SimonHeybrock** [comment]
> Not so sure any more if I like what I said above. The mechanism would be:
> 
>     * Create environment with required scipp version.
> 
>     * Run the "What's New" notebook.
> 
>     * Save with outputs.
> 
>     * Run `make docs` to generate `html`.
> 
>     * Add, commit, and push to https://github.com/scipp/scipp.github.io
> 
>     * **Hope that nothing accidentally removes it**, otherwise, go back to square 1.

We would also need to copy it into the release folder. At this point this is probably not worth it, too messy.... I am going to put the outputs into the main repo for now, until someone has a better idea @nvaytet

**Pull request merged**